bpf/tests/connectivity_netdev/esp.h

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/* Copyright Authors of Cilium */
+
+PKTGEN("tc", "netdev_esp_to_remote_node_ipv4")
+int netdev_esp_to_remote_node_ipv4_pktgen(struct __ctx_buff *ctx)
+{
+	struct pktgen builder;
+	struct iphdr *l3;
+	struct ip_esp_hdr *esp;
+
+	pktgen__init(&builder, ctx);
+
+	l3 = pktgen__push_ipv4_packet(&builder,
+				      (__u8 *)node_mac, (__u8 *)remote_node_mac,
+				      NODE_IP, REMOTE_NODE_IP);
+	if (!l3)
+		return TEST_ERROR;
+
+	esp = pktgen__push_default_esphdr(&builder);
+	if (!esp)
+		return TEST_ERROR;
+
+	esp->spi = REMOTE_NODE_SPI;
+	/* TODO macro-fy this, and validate it */
+	esp->seq_no = 123;
+
+	pktgen__finish(&builder);
+
+	return 0;
+}
+
+SETUP("tc", "netdev_esp_to_remote_node_ipv4")
+int netdev_esp_to_remote_node_ipv4_setup(struct __ctx_buff *ctx)
+{
+	/* TODO is this correct? */
+	ctx->mark = ipsec_encode_encryption_mark(REMOTE_NODE_SPI, REMOTE_NODE_NODE_ID);
```

Comment:
After encryption skb->mark will be reset to 0xe00 (MARK_MAGIC_ENCRYPT) without spi and node_id embedded. This is done by https://github.com/cilium/cilium/pull/28258

Reply:
ah did I nerd-snipe you on this? :grin:  Thank you!